src/grdmix.c

```diff
@@ -0,0 +1,707 @@
+		if ((weights = grdmix_get_array (GMT, &(Ctrl->W), Ctrl->In.type[BLEND], &G_in[BLEND], &I_in[BLEND], h[0], 0.0, 1.0, "blend weight")) == NULL) {
+			if (alpha) gmt_M_free (GMT, alpha);
+			if (intens) gmt_M_free (GMT, intens);
+			Return (GMT_RUNTIME_ERROR);
+		}
+	}
+
+	/* Deal with operation -D up front, then exit */
+
+	if (Ctrl->D.active) {	/* De-construct single image into its gray or R,G,B [plus optional A] layers */
+		char code[4] = {'R', 'G', 'B', 'A'}, file[PATH_MAX] = {""};
+		uint64_t off = 0;
+		float scale = 1.0 / Ctrl->N.factor;	/* To avoid division */
+
+		H = I_in[0]->header;
+		GMT_Report (API, GMT_MSG_INFORMATION, "Deconstruct image into component grid layers\n", Ctrl->In.file[0]);
+		if (H->n_bands == 1) code[0] = 'g';	/* Grayscale */
+		if (gmt_M_360_range (H->wesn[XLO], H->wesn[XHI]) && gmt_M_180_range (H->wesn[YHI], H->wesn[YLO]))
+			gmt_set_geographic (GMT, GMT_IN);	/* If exactly fitting the Earth then we assume geographic image */
+		for (band = 0; band < H->n_bands; band++) {	/* March across the RGB values in both images and increment counters */
+			if ((G = GMT_Create_Data (API, GMT_IS_GRID, GMT_IS_SURFACE, GMT_CONTAINER_AND_DATA, NULL, H->wesn, H->inc, GMT_GRID_PIXEL_REG, 0, NULL)) == NULL) {
+				GMT_Report (API, GMT_MSG_ERROR, "Unable to create a grid for output!\n");
+				Return (GMT_RUNTIME_ERROR);
+			}
+			off = band * H->size;
+#ifdef _OPENMP
+#pragma omp parallel for private(row,col,node) shared(GMT,G,off,scale,I_in)
+#endif
+			gmt_M_grd_loop (GMT, G, row, col, node)
+				G->data[node] = scale * I_in[0]->data[node+off];
+
+			sprintf (file, Ctrl->G.file, code[band]);
+			/* Write out grid */
+			if (GMT_Set_Comment (API, GMT_IS_GRID, GMT_COMMENT_IS_OPTION | GMT_COMMENT_IS_COMMAND, options, G)) {
+				Return (API->error);
+			}
+			if (GMT_Write_Data (API, GMT_IS_GRID, GMT_IS_FILE, GMT_IS_SURFACE, GMT_CONTAINER_AND_DATA, NULL, file, G) != GMT_NOERROR) {
+				Return (API->error);
+			}
+			if (GMT_Destroy_Data (API, &G) != GMT_NOERROR) {
+				GMT_Report (API, GMT_MSG_ERROR, "Unable to free grid!\n");
+				Return (GMT_RUNTIME_ERROR);
+			}
+		}
+		Return (GMT_NOERROR);
+	}
+
+	if (Ctrl->C.active) {	/* Combine 1 or 3 grids into a new single image, while handling the optional -A -I information */
+		uint64_t dim[3] = {0,0,3};
+		GMT_Report (API, GMT_MSG_INFORMATION, "Construct image from %d component grid layers\n", Ctrl->In.n_in);
+		if ((I = GMT_Create_Data (API, GMT_IS_IMAGE, GMT_IS_SURFACE, GMT_CONTAINER_AND_DATA, dim, G_in[0]->header->wesn, G_in[0]->header->inc, GMT_GRID_PIXEL_REG, 0, NULL)) == NULL) {
+			GMT_Report (API, GMT_MSG_ERROR, "Unable to duplicate an image for output!\n");
+			Return (GMT_RUNTIME_ERROR);
+		}
+		H = I->header;
+		if (Ctrl->I.active && Ctrl->In.n_in == 3) {	/* Make the most work-intensive version under OpenMP */
+#ifdef _OPENMP
+#pragma omp parallel for private(row,col,node,band,rgb,pix) shared(GMT,I,G_in,H)
```

Comment:
intens is also a shared variable, right?
```suggestion
#pragma omp parallel for private(row,col,node,band,rgb,pix) shared(GMT,I,G_in,H,intens)
```

Reply:
Yes, fixed.